microbench/src/test/java/io/netty/microbench/http2/Http2FrameWriterBenchmark.java

```diff
@@ -0,0 +1,562 @@
+                serverChannel.close().sync();
+                Future<?> serverGroup = sb.group().shutdownGracefully(0, 0, MILLISECONDS);
+                Future<?> serverChildGroup = sb.childGroup().shutdownGracefully(0, 0, MILLISECONDS);
+                Future<?> clientGroup = cb.group().shutdownGracefully(0, 0, MILLISECONDS);
+                serverGroup.sync();
+                serverChildGroup.sync();
+                clientGroup.sync();
+            }
+        }
+    }
+
+    interface EnvironmentParameters {
+        EventLoopGroup newEventLoopGroup();
+
+        Class<? extends ServerChannel> serverChannelClass();
+
+        Class<? extends Channel> clientChannelClass();
+
+        ByteBufAllocator clientAllocator();
+    }
+
+    private static Environment boostrapEnvWithTransport(final EnvironmentParameters params) {
+        try {
+            final Environment env = new Environment();
+            env.sb = new ServerBootstrap();
+            env.cb = new Bootstrap();
+
+            EventLoopGroup serverEventLoopGroup = params.newEventLoopGroup();
+            env.sb.group(serverEventLoopGroup, serverEventLoopGroup);
+            env.sb.channel(params.serverChannelClass());
+            env.sb.childHandler(new ChannelInitializer<Channel>() {
+                @Override
+                protected void initChannel(Channel ch) throws Exception {
+                }
+            });
+
+            env.cb.group(params.newEventLoopGroup());
+            env.cb.channel(params.clientChannelClass());
+            env.cb.handler(new ChannelInitializer<Channel>() {
+                @Override
+                protected void initChannel(Channel ch) throws Exception {
+                    ChannelPipeline p = ch.pipeline();
+                    Http2Connection connection = new DefaultHttp2Connection(false);
+                    env.clientWriter = new DefaultHttp2FrameWriter();
+                    Http2ConnectionHandler connectionHandler = new Http2ConnectionHandler(
+                            new DefaultHttp2ConnectionDecoder.Builder().connection(connection)
+                                    .frameReader(new DefaultHttp2FrameReader()).listener(new Http2FrameAdapter()),
+                            new DefaultHttp2ConnectionEncoder.Builder().connection(connection).frameWriter(
+                                    env.clientWriter));
+                    p.addLast(connectionHandler);
+                    env.clientContext = p.lastContext();
+                }
+            });
+
+            env.serverChannel = env.sb.bind(new InetSocketAddress(0)).sync().channel();
+            int port = ((InetSocketAddress) env.serverChannel.localAddress()).getPort();
+
+            ChannelFuture ccf = env.cb.connect(new InetSocketAddress(NetUtil.LOCALHOST, port));
+            assertTrue(ccf.awaitUninterruptibly().isSuccess());
+            return env;
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```

Comment:
So I can't re-throw, unless I wrap in a RuntimeException.  Related to your next comment I will handle all exceptions that are un-expected, or would hose the benchmark so that they fail the benchmark.